lib/sedna/README.md

```diff
@@ -1,2 +1,224 @@
+
+    ```
+
+2. Define an `Estimator`:
+
+	```python
+	
+    import os
+    
+    # Keras
+    import keras
+    from keras.layers import Dense, MaxPooling2D, Conv2D, Flatten, Dropout
+    from keras.models import Sequential
+    
+    os.environ['BACKEND_TYPE'] = 'KERAS'
+    
+    def KerasEstimator():
+        model = Sequential()
+        model.add(Conv2D(64, kernel_size=(3, 3),
+                         activation="relu", strides=(2, 2),
+                         input_shape=(128, 128, 3)))
+        model.add(MaxPooling2D(pool_size=(2, 2)))
+        model.add(Conv2D(32, kernel_size=(3, 3), activation="relu"))
+        model.add(MaxPooling2D(pool_size=(2, 2)))
+        model.add(Flatten())
+        model.add(Dropout(0.25))
+        model.add(Dense(64, activation="relu"))
+        model.add(Dense(32, activation="relu"))
+        model.add(Dropout(0.5))
+        model.add(Dense(2, activation="softmax"))
+        
+        model.compile(loss="categorical_crossentropy",
+                      optimizer="adam",
+                      metrics=["accuracy"])
+        loss = keras.losses.CategoricalCrossentropy(from_logits=True)
+        metrics = [keras.metrics.categorical_accuracy]
+        optimizer = keras.optimizers.Adam(learning_rate=0.1)
+        model.compile(loss=loss, optimizer=optimizer, metrics=metrics)
+        return model    
+     ```
+
+    ```python
+    # XGBOOST
+    
+    import os
+    import xgboost
+ 
+    os.environ['BACKEND_TYPE'] = 'SKLEARN'
+ 
+    XGBEstimator = xgboost.XGBClassifier(
+            learning_rate=0.1,
+            n_estimators=600,
+            max_depth=2,
+            min_child_weight=1,
+            gamma=0,
+            subsample=0.8,
+            colsample_bytree=0.8,
+            objective="multi:softmax",
+            num_class=3,
+            nthread=4,
+            seed=27
+     )
+ 
+    ```
+    
+    ```python
+ 
+    # Customize
+ 
+    class Estimator:
+
+        def __init__(self, **kwargs):
+            ...
+            
+        def load(self, model_url=""):
+            ...
+            
+        def save(self, model_path=None):
+            ...
+
+        def predict(self, data, **kwargs):
+            ...
+            
+        def evaluate(self, valid_data, **kwargs):
+            ...
+            
+        def train(self, train_data, valid_data=None, **kwargs):
+            ...
+ 
+	```
+    > **Notes**:  Estimator is a high-level API that greatly simplifies machine learning programming. Estimators encapsulate training, evaluation, prediction, and exporting for your model. 
+
+
+3. Initialize a Incremental Learning Job:
+
+    ```python
+   
+    # get hard exmaple mining algorithm from config
+    hard_example_mining = IncrementalLearning.get_hem_algorithm_from_config(
+        threshold_img=0.9
+    )
+   
+    # create Incremental Learning infernece instance
+    il_job = IncrementalLearning(
+        estimator=Estimator,
+        hard_example_mining=hard_example_mining,
```

Comment:
Is there a redundant comma after "hard_example_mining"?

Reply:
check